src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
+/* eslint-disable no-useless-constructor */
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 export interface Figure {
-
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0
+      || this.c >= this.a + this.b) {
+      throw new Error('Not valid arguments');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semi = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(semi * (semi - this.a)
+      * (semi - this.b) * (semi - this.c));
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shapes.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius < 0) {
+      throw new Error('Not valid arguments');
+    }
+  }
 
+  getArea(): number {
+    const area = Math.PI * (this.radius ** 2);
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape = Shapes.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Not valid arguments');
```

Comment:
Check 3 point in the [checklist](https://github.com/mate-academy/ts_figures_classes/blob/master/checklist.md)